src/NLog/LogManager.cs

```diff
@@ -435,5 +435,42 @@ private static void TurnOffLogging(object sender, EventArgs args)
             Configuration = null;
             InternalLogger.Info("Logger has been shut down.");
         }
+
+        static IFileCompressor _fileCompressor;
```

Comment:
please move those changes to the `FileTarget` (LogManager should not have knowledge of the targets)